### PR TITLE
Day sixteen

### PR DESCRIPTION
**Pull request by Qairkhan**
Добавлена новая станица (Find users). Добавлен axios для запросов на сервер (axios.get) (https://social-network.samuraijs.com/api/1.0/users) . На данной странице использована классовая компонента.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -12,27 +12,19 @@ import Music from "./components/Music/Music";
 import Settigns from "./components/Settings/Settings";
 
 import "./App.css";
```
Comment (Dmitry-White) [inline src/App.js:15]: Consider importing component's styles as the last of the imports.

File: src/App.js [hunk 2]
```diff
@@ -12,27 +12,19 @@ import Music from "./components/Music/Music";
...
+import UsersContainer from "./components/Users/UsersContainer";
```
Comment (Dmitry-White) [inline src/App.js]: Consider grouping this import with the Settings import - they belong to the same folder level

File: src/components/Users/Users.jsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+      .get("https://social-network.samuraijs.com/api/1.0/users")
+      .then((response) => {
+        this.props.setUsers(response.data.items);
+      });
+  }
+
+  render() {
+    return (
+      <div>
+        {this.props.users.map((u) => (
```
Comment (Dmitry-White) [inline src/components/Users/Users.jsx]: Remember - NEVER do `[...].map(...)` inside the "render" method of React Class component.
Consider creating 2 separate functions:
1. "getUsers" should hold the logic of mapping;
2. "getUser" should hold the logic of a map callback.

File: src/components/Users/Users.jsx [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+          <div key={u.id}>
+            <span>
+              <div>
+                <img
+                  src={u.photos.small != null ? u.photos.small : photo000}
+                  className={s.avatar}
+                />
+              </div>
+              <div>
+                {u.followed ? (
```
Comment (Dmitry-White) [inline src/components/Users/Users.jsx]: Consider creating a separate function to hold this JSX logic.

File: src/components/Users/Users.jsx [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+                  <button
+                    onClick={() => {
```
Comment (Dmitry-White) [inline src/components/Users/Users.jsx]: Consider creating a separate function to hold this handler logic.

File: src/components/Users/Users.jsx [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+                      this.props.unfollow(u.id);
+                    }}
+                  >
+                    UnFollow
+                  </button>
+                ) : (
+                  <button
+                    onClick={() => {
```
Comment (Dmitry-White) [inline src/components/Users/Users.jsx]: Consider creating a separate function to hold this handler logic.

File: src/components/Users/UsersContainer.jsx [hunk 7]
```diff
@@ -0,0 +1,33 @@
+import React from "react";
+
+import Users from "./Users";
+import { connect } from "react-redux";
```
Comment (Dmitry-White) [inline src/components/Users/UsersContainer.jsx:2]: This is a third-party library import - move it to the appropriate group.

File: src/components/Users/UsersContainer.jsx [hunk 8]
```diff
@@ -0,0 +1,33 @@
+import React from "react";
+
+import Users from "./Users";
```
Comment (Dmitry-White) [inline src/components/Users/UsersContainer.jsx]: This import should come after the "redux/actionCreators" group

File: src/redux/actionCreators.js [hunk 9]
```diff
@@ -18,9 +18,24 @@ const updateNewMessageBodyCreator = (payload) => {
 
 const updateSendMessageCreator = () => ({ type: ACTION_TYPES.SEND_MESSAGE });
 
+const updateFollowStatusAC = (userId) => ({
+  type: ACTION_TYPES.FOLLOW_TO_USER,
+  userId,
```
Comment (Dmitry-White) [inline src/redux/actionCreators.js]: Consider keeping an action body generic. Every action should have 2 fields: type and payload.
So add "payload" key and move "userId" under it.
`{ type: '...', payload: userId, }`

**Review by Dmitry-White — CHANGES_REQUESTED** [review]

File: src/App.js (hunk 1 above)
Comment (Qairkhan) [reply src/App.js:15]: done

File: src/App.js (hunk 2 above)
Comment (Qairkhan) [reply src/App.js]: done

File: src/components/Users/UsersContainer.jsx (hunk 7 above)
Comment (Qairkhan) [reply src/components/Users/UsersContainer.jsx:2]: done

File: src/components/Users/UsersContainer.jsx (hunk 8 above)
Comment (Qairkhan) [reply src/components/Users/UsersContainer.jsx]: done

File: src/components/Users/Users.jsx (hunk 6 above)
Comment (Qairkhan) [reply src/components/Users/Users.jsx]: done

File: src/components/Users/Users.jsx (hunk 5 above)
Comment (Qairkhan) [reply src/components/Users/Users.jsx]: done

File: src/redux/actionCreators.js (hunk 9 above)
Comment (Qairkhan) [reply src/redux/actionCreators.js]: done

File: src/components/Users/Users.jsx (hunk 4 above)
Comment (Qairkhan) [reply src/components/Users/Users.jsx]: done

File: src/components/Users/Users.jsx (hunk 3 above)
Comment (Qairkhan) [reply src/components/Users/Users.jsx]: done

**Review by Dmitry-White — APPROVED** [review]

**Pull request merged**